src/app/common/router/withRouter.js

```diff
@@ -0,0 +1,61 @@
+import { useContext, useMemo } from 'react'
+import { getRoterConfigs } from './routerConfigs'
+import isEmpty from 'lodash/isEmpty'
+import get from 'lodash/get'
+import pathToRegexp from 'path-to-regexp'
+import { __RouterContext as RouterContext } from 'react-router'
+import { buildQueryParams } from 'common/utils/queryParams'
+
+
+export default function withNamedRouter(ChildComponent) {
+  return function NamedRouter(props) {
+    const routerValue = useContext(RouterContext)
+    const history = useMemo(() => namedHistory(routerValue.history), [routerValue])
+    return (
+      <ChildComponent
+        history={history}
+        location={routerValue.location}
+        match={routerValue.location}
+        {...props}
+      />
+    )
+  }
+}
+
+function namedHistory(location) {
+  return {
+    ...location,
+    push: (path, state) => location.push(namedNavigation(path, state), state),
+    replace: (path, state) => location.replace(namedNavigation(path, state), state),
```

Comment:
we need both options:
- navigate by name
- navigate by path

Reply:
done